drgn/helpers/linux/list_nulls.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Nulls Lists
+-----------
+
+The ``drgn.helpers.linux.list_nulls`` module provides helpers for working with
+the special version of lists (``struct hlist_nulls_head`` and ``struct
+hlist_nulls_node``) in :linux:`include/linux/list_nulls.h`, where end of list
+is not a NULL pointer, but a 'nulls' marker.
+"""
+
+from drgn import container_of
+
+
+__all__ = [
+    'is_a_nulls',
+    'hlist_nulls_empty',
+    'hlist_nulls_entry',
+    'hlist_nulls_for_each_entry',
+]
+
+
+def is_a_nulls(pos):
+    """
+    .. c:function:: bool is_a_nulls(const struct hlist_nulls_node *ptr)
+
+    Return whether a a pointer is a nulls marker.
+    """
+    return bool(pos.value_() & 1)
+
+
+def hlist_nulls_empty(head):
+    """
+    .. c:function:: bool hlist_nulls_empty(const struct hlist_nulls_head *head)
+
+    Return whether a nulls hash list is empty.
+    """
+    return is_a_nulls(head.first)
+
+
+def hlist_nulls_entry(pos, type, member):
+    """
+    .. c:function:: type *hlist_nulls_entry(struct hlist_nulls_node *ptr, type, member)
+
+    Return an entry in a nulls hash list.
+
+    The nulls hash list is assumed to be non-empty.
+    """
+    return container_of(pos, type, member)
+
+
+def hlist_nulls_for_each_entry(head, type, member):
+    """
+    .. c:function:: hlist_nulls_for_each_entry(type *tpos, struct hlist_nulls_node *pos, struct hlist_nulls_head *head, member)
```

Comment:
The C signatures are for documenting the helper itself rather than the original C function/macro it came from, so this could be:

```
.. c:function:: hlist_nulls_for_each_entry(type, struct hlist_nulls_head *head, member)
```

On a related note, the names in the C signatures should be consistent with the names in the Python signature for the other helpers.

Reply:
Yeah, I mistakenly assumed that it's copy/paste of C signature. Makes sense, I'll cleanup all signatures here and in other places.